src/nomad_simulations/schema_packages/properties/electronic_eigenvalues.py

```diff
@@ -0,0 +1,256 @@
+        )
+        sorted_occupation = np.take_along_axis(
+            flattened_occupation, sorted_indices, axis=0
+        )
+        self.m_cache['sorted_eigenvalues'] = True
+        return sorted_value, sorted_occupation
+
+    def resolve_homo_lumo_eigenvalues(
+        self,
+    ) -> tuple[pint.Quantity | None, pint.Quantity | None]:
+        """
+        Resolve the `highest_occupied` and `lowest_unoccupied` eigenvalues by performing a binary search on the
+        flattened and sorted `value` and `occupation`. If these quantities already exist, overwrite them or return
+        them if it is not possible to resolve from `value` and `occupation`.
+
+        Returns:
+            (tuple[Optional[pint.Quantity], Optional[pint.Quantity]]): The `highest_occupied` and
+            `lowest_unoccupied` eigenvalues.
+        """
+        # Sorting `value` and `occupation`
+        ordered_results = self.order_eigenvalues()
+        if ordered_results is not None:
+            sorted_value, sorted_occupation = ordered_results
+            sorted_value_unit = sorted_value.u
+            sorted_value = sorted_value.magnitude
+        else:
+            if self.highest_occupied is not None and self.lowest_unoccupied is not None:
+                return self.highest_occupied, self.lowest_unoccupied
+            return None, None
+
+        # Binary search to find the transition point between `occupation = 2` and `occupation = 0`
+        homo = self.highest_occupied
+        lumo = self.lowest_unoccupied
+        mid = (
+            np.searchsorted(
+                sorted_occupation <= configuration.occupation_tolerance, True
+            )
+            - 1
+        )
+        if mid >= 0 and mid < len(sorted_occupation) - 1:
+            if sorted_occupation[mid] > 0 and (
+                sorted_occupation[mid + 1] >= -configuration.occupation_tolerance
+                and sorted_occupation[mid + 1] <= configuration.occupation_tolerance
+            ):
+                homo = sorted_value[mid] * sorted_value_unit
+                lumo = sorted_value[mid + 1] * sorted_value_unit
+
+        return homo, lumo
+
+    def extract_band_gap(self) -> ElectronicBandGap | None:
+        """
+        Extract the electronic band gap from the `highest_occupied` and `lowest_unoccupied` eigenvalues.
+        If the difference of `highest_occupied` and `lowest_unoccupied` is negative, the band gap `value` is set to 0.0.
+
+        Returns:
+            (Optional[ElectronicBandGap]): The extracted electronic band gap section to be stored in `Outputs`.
+        """
+        band_gap = None
+        homo, lumo = self.resolve_homo_lumo_eigenvalues()
+        if homo and lumo:
+            band_gap = ElectronicBandGap(is_derived=True, physical_property_ref=self)
+
+            if (lumo - homo).magnitude < 0:
+                band_gap.value = 0.0
+            else:
+                band_gap.value = lumo - homo
+        return band_gap
+
+    def normalize(self):
```

Comment:
Pls correct the arity of the arguments to:

```suggestion
    def normalize(self, archive: EntryArchive, logger: Boundlogger):
```

```python
def normalize(self, *args):
```

also works in principle.